src/other/index.jl

```diff
@@ -112,12 +112,12 @@ function Base.insert!(x::Index, idx::Integer, nm::Symbol)
 end
 
 Base.getindex(x::Index, idx::Symbol) = x.lookup[idx]
-Base.getindex(x::AbstractIndex, idx::Real) = @compat Int(idx)
-Base.getindex(x::AbstractIndex, idx::AbstractVector{Nullable{Bool}}) =
-    getindex(x, convert(Vector{Bool}, idx, false))
-Base.getindex{T<:Nullable}(x::AbstractIndex, idx::AbstractVector{T}) =
-    getindex(x, dropnull(idx))
+Base.getindex(x::AbstractIndex, idx::Real) = Int(idx)
+Base.getindex(x::AbstractIndex, idx::AbstractVector{?Bool}) =
+    getindex(x, Bool[ifelse(isnull(x), false, x) for x in idx])
```

Comment:
While you're at it, it would be more efficient to use `find` directly just like for `Bool`. No need to allocate a big vector with many `false` values just to destroy it.